redcap/project.py

```diff
@@ -149,6 +157,423 @@ def _call_api(self, payload, typpe, **kwargs):
+
+        Parameters
+        ----------
+        format : (``'json'``), ``'csv'``, ``'xml'``
+            Return the project report data,
+            csv or xml, ``'df''`` will return a ``pandas.DataFrame``
+        report_id: str
+            report ID
+        raw_or_label : (``'raw'``), ``'label'``,
+            export the raw coded values or labels for the options of
+            multiple choice fields, or both
+        raw_or_label_headers: (``'raw'``), ``'label'``
+            For "csv" & "flat" type only, exports variable/field names (raw)
+            or the field labels (label)
+        export_checkbox_labels : (``False``), ``True``
+            specify whether to export checkbox values as their label on
+            export.
+        df_kwargs : dict
+            Passed to pandas.read_csv to control construction of
+            returned DataFrame
+
+        Returns
+        -------
+        report: list, str, ``pandas.DataFrame``
+            report data from project
+        """
+
+        # Check for dataframe usage
+        ret_format = self.__toggle_dataframe(format)
+
+        pl = self.__basepl('report')
+
+        to_add = (report_id,raw_or_label,raw_or_label_headers,export_checkbox_labels)
+        str_add = ('report_id','rawOrLabel','rawOrLabelHeadHeaders','exportCheckboxLabel')
+        for key, data in zip(str_add, to_add):
+            if data:
+                pl[key] = data
+
+        response, _ = self._call_api(pl, 'exp_report')        
+
+        if format in ('json', 'csv', 'xml'):
+            return response
+        elif format == 'df':
+            if not df_kwargs:
+                return read_csv(StringIO(response))
+            else:
+                return read_csv(StringIO(response), **df_kwargs)  
+
+    def export_instruments(self, format='json',df_kwargs=None):
+        """
+        Export the project's instruments
+
+        Parameters
+        ----------
+        format : (``'json'``), ``'csv'``, ``'xml'``
+            Return the list of project instruments in native objects,
+            csv or xml, ``'df''`` will return a ``pandas.DataFrame``
+        df_kwargs : dict
+            Passed to pandas.read_csv to control construction of
+            returned DataFrame
+
+        Returns
+        -------
+        instruments: list, str, ``pandas.DataFrame``
+            list of data collection instruments
+        """
+        # Check for dataframe usage
+        ret_format = self.__toggle_dataframe(format)
+
+        pl = self.__basepl('instrument')
+        response, _ = self._call_api(pl, 'exp_instrument')
+
+        if format in ('json', 'csv', 'xml'):
+            return response
+        elif format == 'df':
+            if not df_kwargs:
+                return read_csv(StringIO(response))
+            else:
+                return read_csv(StringIO(response), **df_kwargs)   
+
+    def export_pdf(self, format='json',record=None,event=None,instrument=None,all_records=None,df_kwargs=None):
```

Comment:
I'd have to double-check, and I'm having some hiccups rebuilding my vagrant REDCap instance at the moment, but I believe it still accepts a format argument. 

If I recall correctly, the error message will be handled to that format, which might be useful depending on if more elaborate error-handling is added later.